File: app/src/main/java/com/qingyii/hxtz/wmcj/mvp/ui/fragment/WorkParkFragment.java
package com.qingyii.hxtz.wmcj.mvp.ui.fragment;

import android.content.Intent;
import android.os.Bundle;
import android.os.Message;
import android.support.v7.widget.DefaultItemAnimator;
import android.support.v7.widget.DividerItemDecoration;
import android.support.v7.widget.LinearLayoutManager;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.jess.arms.base.BaseFragment;
import com.jess.arms.di.component.AppComponent;
import com.jess.arms.widget.autolayout.AutoTabLayout;
import com.mcxtzhang.commonadapter.rv.HeaderFooterAdapter;
import com.qingyii.hxtz.R;
import com.qingyii.hxtz.base.app.EventBusTags;
import com.qingyii.hxtz.base.widget.AutoLoadMoreRecyclerView;
import com.qingyii.hxtz.wmcj.WMCJContract;
import com.qingyii.hxtz.wmcj.di.component.DaggerWorkComponent;
import com.qingyii.hxtz.wmcj.di.module.WorkModule;
import com.qingyii.hxtz.wmcj.mvp.model.bean.WorkParkbean;
import com.qingyii.hxtz.wmcj.mvp.presenter.WorkParkItemPresenter;
import com.qingyii.hxtz.wmcj.mvp.presenter.WorkPresenter;
import com.qingyii.hxtz.wmcj.mvp.ui.activity.WorkParkCategoryActivity;


import org.simple.eventbus.Subscriber;
import org.simple.eventbus.ThreadMode;

import java.util.ArrayList;


import javax.inject.Inject;

import butterknife.BindView;
import in.srain.cube.views.ptr.PtrClassicFrameLayout;
import in.srain.cube.views.ptr.PtrDefaultHandler;
import in.srain.cube.views.ptr.PtrFrameLayout;

/**
 * Created by zhf on 2018/3/12.  改写后的文明创建工作动态
 */

public class WorkParkFragment extends BaseFragment<WorkPresenter>  implements WMCJContract.WorkParkView {
    @BindView(R.id.toolbar_back)
    Button back;

    @BindView(R.id.toolbar_title)
    TextView title;

    @BindView(R.id.toolbar_right)
    Button  zhuanti;

    @BindView(R.id.workpark_tablayout)
    AutoTabLayout tablayout;

   @BindView(R.id.workpark_item_recyclerView)
    AutoLoadMoreRecyclerView recyclerView;

    @BindView(R.id.workpark_item_refresh)
    PtrClassicFrameLayout ptr;

    private ArrayList<WorkParkbean.DataBean.MenuItemBean> list1=new ArrayList<>();
    private ArrayList<WorkParkbean.DataBean.TagItemBean>  list2=new ArrayList<>();
    //private ArrayList<Fragment>  list=new ArrayList<>();
   // private WorkparkvpAdapter adapter;

    @Inject
    WorkParkItemPresenter mPresenter1;

    private boolean flag=true;
    private int system_id=-99;
    private int id;


    @Override
    public void setupFragmentComponent(AppComponent appComponent) {
        DaggerWorkComponent.builder()
                .appComponent(appComponent)
                .workModule(new WorkModule(this))
                .build()
                .inject(this);
    }

    @Override
    public View initView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_workpark,container,false);
    }

    @Override
    public void initData(Bundle savedInstanceState) {
       initview();
       mPresenter1.initadapter();

        ptr.setLastUpdateTimeRelateObject(this);
        ptr.setPtrHandler(new PtrDefaultHandler() {
            @Override
            public void onRefreshBegin(PtrFrameLayout frame) {

                if(system_id!=-99){
                    mPresenter1.getdata(id,system_id);
                } else {
                    mPresenter1.getdata(id);
                }

                if(flag){
                    mPresenter1.getimages();
                }
            }

            @Override
            public boolean checkCanDoRefresh(PtrFrameLayout frame, View content, View header) {
                return super.checkCanDoRefresh(frame, content, header);
            }
        });

        recyclerView.setOnLoadMoreListener(new AutoLoadMoreRecyclerView.OnLoadMoreListener() {
            @Override
            public void loadMore() {
                if(system_id!=-99){
                    mPresenter1.getdatamore(id,system_id);
                }  else {
                    mPresenter1.getdatamore(id);
                }
            }
        });


    }

    private void initview() {
        back.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getActivity().finish();
            }
        });
        title.setText("工作动态");
        zhuanti.setText("专题");

        zhuanti.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent  intent=new Intent(getActivity(), WorkParkCategoryActivity.class);
                startActivity(intent);
            }
        });

        mPresenter.getWorkMenu();
    }

    @Override
    public void setData(Object data) {

    }


    @Override
    public void showLoading() {
        recyclerView.notifyMoreLoaded();
    }

    @Override
    public void hideLoading() {
     ptr.refreshComplete();
    }

    @Override
    public void showMessage(String message) {

    }

    @Override
    public void launchActivity(Intent intent) {

    }

    @Override
    public void killMyself() {

    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        mPresenter1.onDestroy();
        list1.clear();
        list2.clear();
    }

    @Override
    public void getdatasuccess(WorkParkbean workParkbean) {
        zhuanti.setVisibility(View.VISIBLE);
        list1.clear();
        list1.addAll(workParkbean.getData().getMenu_item());
        list2.clear();
        list2.addAll(workParkbean.getData().getTag_item());
        id=list1.get(0).getId();
        system_id=list1.get(0).getSystem_id();
        getdata();
    }

    //
    private void getdata() {
        if(system_id!=-99){
            mPresenter1.getdata(id,system_id);
        }
        else {
            Log.i("tmdvipager","----"+id);
            mPresenter1.getdata(id);
        }
        if(flag){
            Log.i("tmddiyige","------");
            mPresenter1.getimages();

        }

    }

    @Override
    public void getdataerror(Exception e) {

    }

    @Override
    public void setRecyclerviewAdapter(HeaderFooterAdapter adapter) {
         recyclerView.setAutoLayoutManager(new LinearLayoutManager(getActivity()))
                .setAutoItemAnimator(new DefaultItemAnimator())
                .addAutoItemDecoration(new DividerItemDecoration(getActivity(), LinearLayout.VERTICAL))
                .setAutoAdapter(adapter);
    }

    @Override
    public void doRefresh() {
        ptr.autoRefresh();
    }

    @Override
    public void finishRefresh() {
        ptr.refreshComplete();
    }

    @Override
    public void setflag(boolean flag) {

    }

    @Subscriber(mode = ThreadMode.MAIN, tag = EventBusTags.WORKPARK)
    public void getzhuanti(Message msg){
       recyclerView.scrollToPosition(0);
        int i=msg.arg1;
       if(i!=0){
           flag=false;
          mPresenter1.clearhead();
       } else {
           flag= true;
       }
        if(i<list1.size()){
            id=list1.get(i).getId();
            system_id=list1.get(i).getSystem_id();
        } else {
            id=list2.get(i-list1.size()).getId();
            system_id=-99;
        }
        getdata();
    }


}

File: app/src/main/java/com/qingyii/hxt/wmcj/mvp/ui/WorkParkFragment.kt
package com.qingyii.hxt.wmcj.mvp.ui

import android.content.Intent
import android.os.Bundle
import android.os.Message
import android.support.v4.app.Fragment
import android.support.v4.view.ViewPager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.LinearLayout
import android.widget.TextView
import com.jess.arms.base.BaseFragment
import com.jess.arms.di.component.AppComponent
import com.jess.arms.mvp.BasePresenter
import com.jess.arms.mvp.IModel
import com.jess.arms.mvp.IView
import com.jess.arms.widget.autolayout.AutoTabLayout
import com.qingyii.hxt.R
import com.qingyii.hxt.base.app.EventBusTags
import com.qingyii.hxt.base.app.GlobalConsts
import com.qingyii.hxt.base.app.bindView
import com.qingyii.hxt.meeting.mvp.ui.fragment.MeetingFragment
import com.qingyii.hxt.wmcj.mvp.model.entity.WorkParkData
import com.qingyii.hxt.wmcj.mvp.ui.adapter.MYViewPagerAdapter
import org.simple.eventbus.Subscriber
import org.simple.eventbus.ThreadMode
import java.util.*

/**
 * 工作动态，workpark为易新煌命名
 * Created by xubo on 2017/9/7.
 */
class WorkParkFragment : BaseFragment<BasePresenter<IModel, IView>>() {
    val toolbarBack: Button by bindView(R.id.toolbar_back)
    val toolbarBackLayout: LinearLayout by bindView(R.id.toolbar_back_layout)
    val toolbarTitle: TextView by bindView(R.id.toolbar_title)
    val toolbarRightTv: TextView by bindView(R.id.toolbar_right_tv)
    val toolbarRight: Button by bindView(R.id.toolbar_right)
    val toolbarRightLayout: LinearLayout by bindView(R.id.toolbar_right_layout)
    val tabLayout: AutoTabLayout by bindView(R.id.workpark_tablayout)
    val viewPager: ViewPager by bindView(R.id.workpark_viewpager)

    companion object {
        var title: String? = null
        var titlesList = ArrayList<WorkParkData>()
        private var inflater: LayoutInflater? = null
    }

    private var adapter: MYViewPagerAdapter? = null

    private var fragmentList: MutableList<Fragment> = ArrayList()

    override fun initData(savedInstanceState: Bundle?) {
        //初始化inflater
        inflater = LayoutInflater.from(activity)
        toolbarRightTv.visibility = View.VISIBLE
        toolbarRightTv.text = "专题"
        //0栏目，1专题
        toolbarRightLayout.setTag(TAG.hashCode(), 0)
        toolbarTitle.text = title
        initColumn()
        tabLayout.setupWithViewPager(viewPager)
        toolbarRightLayout.setOnClickListener {
            when (toolbarRightLayout.getTag(TAG.hashCode())) {
            //当前页面为栏目,按钮显示为专题
                0 -> {
                    activity.startActivity(Intent(activity, WorkParkCategoryActivity::class.java))
                }
            //当前页面为专题，按钮显示为返回栏目
                1 -> {
                    initColumn()
                    toolbarRightTv.text = "专题"
                    toolbarRightLayout.setTag(TAG.hashCode(), 0)
                }
            }
        }
        toolbarBackLayout.setOnClickListener { activity.finish() }
    }

    private fun initColumn() {
        titlesList.clear()
        titlesList.add(WorkParkData(0, "栏目一级"))
        titlesList.add(WorkParkData(1, "栏目二"))
        titlesList.add(WorkParkData(2, "栏目三"))
        fragmentList.clear()
        viewPager.removeAllViews()
        for (i in titlesList.indices) {
            fragmentList.add(WorkParkItemFragment())
        }
        if (adapter == null) {
            adapter = MYViewPagerAdapter(context, childFragmentManager, fragmentList, titlesList)
            viewPager.adapter = adapter
        }
        adapter!!.notifyDataSetChanged()
        viewPager.offscreenPageLimit = titlesList.size
        viewPager.currentItem = 0
    }

    override fun setData(data: Any?) {
    }

    override fun setupFragmentComponent(appComponent: AppComponent?) {
        title = arguments.getString(GlobalConsts.TITLE)
    }

    override fun initView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater?.inflate(R.layout.fragment_workpark, container, false)
    }

    override fun onDestroy() {
        super.onDestroy()
        viewPager.removeAllViews()
    }


    @Subscriber(mode = ThreadMode.MAIN, tag = EventBusTags.WORKPARK)
    fun onEventBus(msg: Message) {
        when (msg.what) {
            EventBusTags.WORKPARK_TOPIC -> {
                titlesList.clear()
                titlesList.addAll(msg.obj as List<WorkParkData>)
                fragmentList.clear()
                viewPager.removeAllViews()
//                viewPager.offscreenPageLimit = titlesList.size
                titlesList.forEach {
                    fragmentList.add(MeetingFragment())
                }
                adapter!!.notifyDataSetChanged()
                viewPager.offscreenPageLimit = titlesList.size
                viewPager!!.currentItem = msg.arg1
                toolbarRightTv.text = "返回栏目"
                toolbarRightLayout.setTag(TAG.hashCode(), 1)
            }
        }
    }

}